staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/validation.go

```diff
@@ -145,7 +147,10 @@ func (s *Validator) validateExpressions(ctx context.Context, fldPath *field.Path
 	if s.isResourceRoot {
 		sts = model.WithTypeAndObjectMeta(sts)
 	}
-	activation := NewValidationActivation(obj, sts)
+	var activation interpreter.Activation = NewValidationActivation(ScopedVarName, obj, sts)
+	if oldObj != nil {
+		activation = interpreter.NewHierarchicalActivation(activation, NewValidationActivation(OldScopedVarName, oldObj, sts))
```

Comment:
my read of using NewHierarchicalActivation for this is that it makes every access to `self` slightly more expensive... is that correct?

Reply:
Yea, that's right. I've consolidate everywhere down to a single activation.